docs/REST API/rest-api.md

```diff
@@ -7711,11 +7821,176 @@ Deletes the session specified by ID. When the user associated with the session n
+--------------------------d8c247122f594ba0
+```
+
+##### Default response
+
+`Status: 200`
+
+
+### Download software
+
+_Available in Fleet Premium_
+
+Download software package.
+
+`GET /api/v1/fleet/software/package/:id?alt=media`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                      |
+| ----            | ------- | ---- | --------------------------------------------     |
+| id              | integer | path | **Required**. The ID of the software title to download software package.|
+| alt             | integer | path | **Required**. If specified and set to "media", downloads the specified software package. |
+
+#### Example
+
+`GET /api/v1/fleet/software/package/123?alt=media`
+
+##### Default response
+
+`Status: 200`
+
+```http
+Status: 200
+Content-Type: application/octet-stream
+Content-Disposition: attachment
+Content-Length: <length>
+Body: <blob>
+```
+
+### Delete software
+
+_Available in Fleet Premium_
+
+Delete software package.
+
+`DELETE /api/v1/fleet/software/package/:id`
+
+#### Parameters
+
+| Name            | Type    | In   | Description                                      |
+| ----            | ------- | ---- | --------------------------------------------     |
+| id              | integer | path | **Required**. The ID of the software title to delete software package. |
+
+#### Example
+
+`DELETE /api/v1/fleet/software/package/24`
+
+##### Default response
+
+`Status: 204`
+
+### Get installation results
+
+_Available in Fleet Premium_
+
+`GET /api/v1/fleet/software/install/results`
+
+Get an installation result of the software that was installed.
+
+| Name            | Type    | In   | Description                                      |
+| ----            | ------- | ---- | --------------------------------------------     |
+| installation_uuid | string | path | **Required**. The installation UUID of the software.|
+
+#### Example
+
+`GET /api/v1/fleet/software/install/results`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  {
+    "install_uuid": "b15ce221-e22e-4c6a-afe7-5b3400a017da",
+    "software_title": "Falcon.app",
+    "software_title_id": 8353,
+    "software_package": "FalconSensor-6.44.pkg",
+    "host_id": 123,
+    "host_display_name": "Marko's MacBook Pro",
+    "status": "Failed",
+    "detail": "Script after install failed",
+    "output": "Marko's MacBook Pro",
+    "pre_install_query_output": "Query returned result\nSuccess",
+    "post_install_script_output": "Installing software...\nSuccess\nRolling back software install...\nSuccess"
+  },
+  {
+    "install_uuid": "b15ce221-e22e-4c6a-afe7-5b3400a017da",
+    "software_title": "Zoom.app",
+    "software_title_id": 4353,
+    "software_package": "ZoomInstallerFull.msi",
+    "host_id": 124,
+    "host_display_name": "WindowsLaptop-345034",
+    "status": "Pending",
+    "detail": "",
+    "output": "",
+    "pre_install_query_output": "",
+    "post_install_script_output": ""
+  },
```

Comment:
@noahtalerman, I think there's a design bug here. Shouldn't this endpoint only return one result? This example includes a result for Zoom and Falcon.

Reply:
@gillespi314 That's correct. It should always return one item. I updated PR.